platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/PlatformConstructor.java

```diff
@@ -171,7 +171,7 @@ static SwirldStateManager swirldStateManager(
             @NonNull final PostConsensusSystemTransactionManager postConsensusSystemTransactionManager,
             @NonNull final Metrics metrics,
             @NonNull final PlatformStatusStateMachine platformStatusStateMachine,
-            @NonNull final SettingsProvider settings,
+            @NonNull final TransactionConfig transactionConfig,
```

Comment:
Instead of passing this in as a parameter, could we get it from the context? No need to extract it in this helper method (which I will probably delete soon, it doesn't really serve a purpose), you can get it inside the `SwirldStateManagerImpl`.

Come to think of it, you could also do the same with metrics as well.

For these big objects with a millions parameters, I think it is worth a little extra logic in the constructor if it reduces the number of parameters.

Reply:
Fixed